cloud/store/service.go

```diff
@@ -208,12 +208,26 @@ func (st *Service) Execute(req *cmd.ApplyRequest) error {
 	if leader == "" {
 		return ErrLeaderNotFound
 	}
-	_, err := st.cl.Apply(leader, req)
+
+	// If we are a follower, ensure we retry up to 3 times requests to the leader with a small delay.
+	// This can help when we have schema inconsistencies or network issues
+	var err error
+	var errCount int
+	for {
+		if _, err := st.cl.Apply(leader, req); err != nil {
+			errCount += 1
+		} else {
+			break
+		}
+
```

Comment:
The one thing I don't know here is if we want to introduce a sleep or a backoff mechanism when retrying on error.

Also it would probably make sense to have an error filter to ensure we don't retry three times requests that are doomed to fail. Not sure how we could detect between the two.

Reply:
i think a back off would be good, i've used the [package](https://github.com/cenkalti/backoff) before and it's has nice backoff mechanism